File: VB/Scaffolding.ValidationErrors/DepartmentContextDataModel/IDepartmentContextUnitOfWork.vb
Imports DevExpress.Mvvm.DataModel
Imports Scaffolding.ValidationErrors.Model
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace Scaffolding.ValidationErrors.DepartmentContextDataModel

	''' <summary>
	''' IDepartmentContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
	''' </summary>
	Public Interface IDepartmentContextUnitOfWork
		Inherits IUnitOfWork

		''' <summary>
		''' The Department entities repository.
		''' </summary>
		ReadOnly Property Departments() As IRepository(Of Department, Integer)
	End Interface
End Namespace

File: CS/Scaffolding.ValidationErrors/DepartmentContextDataModel/IDepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.ValidationErrors.Common.Utils;
using Scaffolding.ValidationErrors.Common.DataModel;
using Scaffolding.ValidationErrors.Common.DataModel.DesignTime;
using Scaffolding.ValidationErrors.Common.DataModel.EntityFramework;
using Scaffolding.ValidationErrors.Model;

namespace Scaffolding.ValidationErrors.DepartmentContextDataModel {
    /// <summary>
    /// IDepartmentContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    /// </summary>
    public interface IDepartmentContextUnitOfWork : IUnitOfWork {

        /// <summary>
        /// The Department entities repository.
        /// </summary>
        IRepository<Department, int> Departments { get; }
    }
}
